### ISSUE TITLE
[BUG] verbosity=4 doesn't work

### ISSUE THREAD

**Issue by zhiqiangdon** [autogluon/autogluon#2878]
- [x] I have checked that this bug exists on the latest stable version of AutoGluon
- [x] and/or I have checked that this bug exists on the latest mainline of AutoGluon via source installation

**Describe the bug**
The logger doesn't work as the verbosity says. AutoGluon code is the latest master branch.

**Expected behavior**
Setting verboity=4 should print the debugging info.

**To Reproduce**
```
train_data = load_pd.load('https://autogluon-text.s3-accelerate.amazonaws.com/glue/sst/train.parquet')
train_data = train_data.sample(n=1000, random_state=0)
predictor = MultiModalPredictor(label='label', eval_metric='acc', verbosity=4)
predictor.fit(train_data, time_limit=60)
```
Calling `fit()` should print some debugging info.

**Screenshots**

**Installed Versions**
Which version of AutoGluon are you are using?  
If you are using 0.4.0 and newer, please run the following code snippet:
<details>

0.6.3b20230204

</details>

**Additional context**

**Comment by zhiqiangdon** [comment]
I think this issue is from `set_logger_verbosity` imported from ` autogluon.common.utils.log_utils`

**Comment by Innixma** [comment]
@zhiqiangdon Can you please provide a minimal code snippet of what logger calls you'd like to have logged at verbosity=4 that are not?

**Comment by zhiqiangdon** [comment]
Here is what expect to see:

```
No path specified. Models will be saved in: "AutogluonModels/ag-20230209_215635/"                   
DEBUG:automm:save path: /home/ubuntu/projects/automm-scripts/AutogluonModels/ag-20230209_215635              
DEBUG:automm:column_types: OrderedDict([('sentence', 'text'), ('label', 'categorical')])            
DEBUG:automm:image columns: []                                                                                       
DEBUG:automm:overrides: None                                                                        
DEBUG:automm:Process col "sentence" with type "text"                                                          
DEBUG:automm:Process col "label" with type label                                                    
DEBUG:automm:selected models: ['hf_text']                                                                                                                                    
DEBUG:automm:model dtypes: ['text']                                                                                                                                          
DEBUG:automm:output_shape: 2                                                                                  
DEBUG:automm:initializing google/electra-base-discriminator                                         
DEBUG:automm:outer layers are treated as head: ['head.weight', 'head.bias']                                     
DEBUG:automm:text max length: 512                                                                   
DEBUG:automm:text segment num: 2                                                                                    
DEBUG:automm:data_processors_count: {'text': 1, 'label': 1}                                         
DEBUG:automm:pos_label: 1                                                                                         
DEBUG:automm:validation_metric_name: val_acc                                                        
DEBUG:automm:minmax_mode: max                                                                                   
Auto select gpus: [0]                                                                               
Using 16bit native Automatic Mixed Precision (AMP)                                                                            
GPU available: True (cuda), used: True                                                              
TPU available: False, using: 0 TPU cores                                                                     
IPU available: False, using: 0 IPUs                                                                 
HPU available: False, using: 0 HPUs                                                                                 
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]                                                           
DEBUG:automm:applying layerwise learning rate decay...                                                             
DEBUG:automm:trainer.max_steps: -1                                                                  
DEBUG:automm:len(trainer.datamodule.train_dataloader()): 100                                                              
DEBUG:automm:trainer.max_epochs: 10          
DEBUG:automm:trainer.accumulate_grad_batches: 16                                                             
DEBUG:automm:max steps: 62                                
DEBUG:automm:warmup steps: 6                                                                                         
DEBUG:automm:lr_schedule: cosine_decay                                                       
DEBUG:automm:done configuring optimizer and scheduler 
```

**Comment by Innixma** [comment]
@zhiqiangdon What I meant was a minimal code example that removes unnecessary complexity from the comparison. I've gone ahead and provided the minimal code example below.

Here we see that multimodal's logger does indeed not update its log level correctly:

```python3

from autogluon.common.utils.log_utils import set_logger_verbosity
from autogluon.multimodal.predictor import logger
# same result if the following is used:
# import logging
# logger = logging.getLogger("automm")

logger.warning('before verbosity 4')
logger.debug("should not see debug")
logger.warning("should see warning")

set_logger_verbosity(verbosity=4, logger=logger)

logger.warning('after verbosity 4')
logger.debug("should see debug")
logger.warning("should see warning")

```

Out:

```
before verbosity 4
should see warning
after verbosity 4
should see warning
```

However, Tabular's logger works correctly:

```python3
from autogluon.common.utils.log_utils import set_logger_verbosity
from autogluon.tabular.predictor.predictor import logger

logger.warning('before verbosity 4')
logger.debug("should not see debug")
logger.warning("should see warning")

set_logger_verbosity(verbosity=4, logger=logger)

logger.warning('after verbosity 4')
logger.debug("should see debug")
logger.warning("should see warning")
```

Out:

```
before verbosity 4
should see warning
after verbosity 4
should see debug
should see warning
```

That's because Tabular's logger is initialized via `logger = logging.getLogger(__name__)`

However, Multimodal's logger is initialized via `logger = logging.getLogger('automm')`

`automm` is not a recognized name space for AutoGluon, and thus is not part of the overall namespace log formatting that is done in `autogluon.common.utils.log_utils._add_stream_handler()` called during initialization of `autogluon.core`

You can fix by instead changing Multimodal's logger to be initialized via `logger = logging.getLogger(__name__)` or `logger = logging.getLogger(anything_that_starts_with_the_string_autogluon)`.

I would suggest using `__name__` unless you have a particular reason why you would want a different value.

**Comment by zhiqiangdon** [comment]
Thanks @Innixma for the investigation! Using `__name__` can't make the verbosity propagate from `predictor.py` to all other files. Making the logger name start with `autogluon` works.

**Comment by Innixma** [comment]
I'd recommend sending a PR with that change, and then using `__name__` for the multimodal loggers.

**Comment by Innixma** [comment]
For example, the way you are currently doing things will not update the verbosity of loggers that come from `autogluon.core` or `autogluon.common`, which you probably do want to update the verbosity of in your situation.

**Comment by Innixma** [comment]
Resolved in #2894, thanks @zhiqiangdon!
